src/DateTime.swift

```diff
@@ -145,16 +145,16 @@ public struct DateTime: Comparable
 
     :returns: A new instance of DateTime.
     */
-    public init(date: NSDate, timezone: NSTimeZone = NSTimeZone.defaultTimeZone())
+    public init(date: Date, timezone: TimeZone = TimeZone.current)
     {
         self.internalDate = date
         self.timezone = timezone
 
-        let calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
+        var calendar = Calendar(identifier: Calendar.Identifier.gregorian)
         calendar.timeZone = timezone
 
-        let calendarUnits = NSCalendarUnit(rawValue: UInt.max)
-        self.calendarComponents = calendar.components(calendarUnits, fromDate: self.internalDate)
+        let calendarUnits = Set<Calendar.Component>([.day, .hour, .year, .minute, .second, .weekday, .weekdayOrdinal, .weekOfMonth, .weekOfYear])
```

Comment:
whaatt why not? let me enums be numbers, man.

Anyway, since I'm not doing much swift these days, this looks reasonable to me as I dont' really have a good idea of how else to approach this. So long as it works, it's fine, and we can review again later